src/Normalizer.php

```diff
@@ -135,45 +137,44 @@ public function convertToString($data): ?string {
 	 *
 	 * We're returning an array here to detect failure because null is a scalar and so is false
 	 *
-	 * @param $data
+	 * @param mixed $data
 	 *
-	 * @return mixed
+	 * @return string|mixed|null
 	 */
 	private function normalizeScalar($data) {
-		if (null === $data || is_scalar($data)) {
-			/*// utf8_encode only works for Latin1 so we rely on mbstring
-			if (is_string($data)) {
-				$data = mb_convert_encoding($data, "UTF-8");
-			}*/
-
-			if (is_float($data)) {
-				$data = $this->normalizeFloat($data);
-			}
+		if ($data === null) {
+			return null;
+		}
+
+		if (is_float($data)) {
+			return $this->normalizeFloat($data);
+		}
 
+		if (is_scalar($data)) {
 			return $data;
 		}
 
-		return [];
+		return null;
```

Comment:
change from empty array to null is intended?

Reply:
yes, I simplified that to work like any other normalizer method. they all return `null` when they didn't handle the `$data`